ambry-router/src/test/java/com.github.ambry.router/GetOperationTest.java

```diff
@@ -0,0 +1,428 @@
+      for (RequestInfo requestInfo : requestListToFill) {
+        ResponseInfo fakeResponse =
+            new ResponseInfo(requestInfo.getRequest(), NetworkClientErrorCode.NetworkError, null);
+        op.handleResponse(fakeResponse);
+        if (op.isOperationComplete()) {
+          break;
+        }
+      }
+      requestListToFill.clear();
+    }
+
+    // At this time requests would have been created for all replicas, as none of them were delivered,
+    // and cross-colo proxying is enabled by default.
+    Assert.assertEquals("Must have attempted sending requests to all replicas", replicasCount,
+        correlationIdToGetOperation.size());
+    Assert.assertTrue("Operation should be complete at this time", op.isOperationComplete());
+    RouterException routerException = (RouterException) op.getOperationException();
+    Assert.assertEquals(RouterErrorCode.OperationTimedOut, routerException.getErrorCode());
+  }
+
+  /**
+   * Test the case where every server returns Blob_Not_Found. All servers must have been contacted,
+   * due to cross-colo proxying.
+   * @throws Exception
+   */
+  @Test
+  public void testBlobNotFoundCase()
+      throws Exception {
+    GetBlobInfoOperation op =
+        new GetBlobInfoOperation(routerConfig, mockClusterMap, responseHandler, blobIdStr, operationFuture, null, time);
+    ArrayList<RequestInfo> requestListToFill = new ArrayList<>();
+    requestRegistrationCallback.requestListToFill = requestListToFill;
+
+    for (MockServer server : mockServerLayout.getMockServers()) {
+      server.setServerErrorForAllRequests(ServerErrorCode.Blob_Not_Found);
+    }
+
+    while (!op.isOperationComplete()) {
+      op.poll(requestRegistrationCallback);
+      List<ResponseInfo> responses = sendAndWaitForResponses(requestListToFill);
+      for (ResponseInfo response : responses) {
+        op.handleResponse(response);
+        if (op.isOperationComplete()) {
+          break;
+        }
+      }
+    }
+
+    Assert.assertEquals("Must have attempted sending requests to all replicas", replicasCount,
+        correlationIdToGetOperation.size());
+    Assert.assertTrue("Operation should be complete at this time", op.isOperationComplete());
+    RouterException routerException = (RouterException) op.getOperationException();
+    Assert.assertEquals(RouterErrorCode.BlobDoesNotExist, routerException.getErrorCode());
+  }
+
+  /**
+   * Test the case with Blob_Not_Found errors from most servers, and Blob_Deleted at just one server. The latter
+   * should be the exception received for the operation.
+   * @throws Exception
+   */
+  @Test
+  public void testErrorPrecedenceWithBlobDeletedAndExpiredCase()
+      throws Exception {
+    ServerErrorCode[] serverErrorCodesToTest = {ServerErrorCode.Blob_Deleted, ServerErrorCode.Blob_Expired};
+    RouterErrorCode[] routerErrorCodesToExpect = {RouterErrorCode.BlobDeleted, RouterErrorCode.BlobExpired};
+    for (int i = 0; i < serverErrorCodesToTest.length; i++) {
```

Comment:
We are testing with a set of `ServerErrorCode`s for which we expect a set of `RouterErrorCode`s. What is the confusion? I thought of creating a class with fields for both, but I think all that is an overkill for a simple test.